R/FilterStateChoices.R

```diff
@@ -477,39 +491,34 @@ ChoicesFilterState <- R6::R6Class( # nolint
             NULL
           })
 
-          if (private$is_checkboxgroup()) {
-            private$observers$selection <- observeEvent(
-              ignoreNULL = FALSE, # it's possible that nothing is selected
-              ignoreInit = TRUE, # ignoreInit: should not matter because we set the UI with the desired initial state
-              eventExpr = input$selection,
-              handlerExpr = {
-                logger::log_trace(sprintf(
-                  "ChoicesFilterState$server@2 selection of variable %s changed, dataname: %s",
-                  private$varname,
-                  private$dataname
-                ))
-                selection <- if (is.null(input$selection)) character(0) else input$selection
-                private$set_selected(selection)
-              }
-            )
-          } else {
-            private$observers$selection <- observeEvent(
-              ignoreNULL = FALSE, # it's possible that nothing is selected
-              ignoreInit = TRUE, # ignoreInit: should not matter because we set the UI with the desired initial state
-              eventExpr = input$selection_open,
-              handlerExpr = {
-                if (!isTRUE(input$selection_open)) {
-                  logger::log_trace(sprintf(
-                    "ChoicesFilterState$server@2 selection of variable %s changed, dataname: %s",
-                    private$varname,
-                    private$dataname
-                  ))
-                  selection <- if (is.null(input$selection)) character(0) else input$selection
-                  private$set_selected(selection)
-                }
+          private$observers$selection <- observeEvent(
+            ignoreNULL = FALSE,
+            ignoreInit = TRUE, # ignoreInit: should not matter because we set the UI with the desired initial state
+            eventExpr = input$selection,
+            handlerExpr = {
+              logger::log_trace(sprintf(
+                "ChoicesFilterState$server@2 selection of variable %s changed, dataname: %s",
+                private$varname,
+                private$dataname
+              ))
+
+              selection <- if (is.null(input$selection) && private$multiple){
+                character(0)
+              } else if (is.null(input$selection) && !private$multiple) {
+                showNotification("single selection is required, choice is ignored")
```

Comment:
feel free to change this message to anything which will help user to understand what should be done 
```suggestion
                showNotification("Choice is ignored as this filter is restricted to single selection only.")
```